### ISSUE TITLE
Move to new BinarySensor base class [HA 0.109]

### ISSUE THREAD

**Issue by mammuth** [mammuth/ha-fritzbox-tools#87]
```
WARNING (MainThread) [homeassistant.components.binary_sensor] BinarySensorDevice is deprecated, modify FritzBoxConnectivitySensor to extend BinarySensorEntity
```

I guess the first version supporting the new `BinarySensorEntity` is 0.109.

**Comment by sagitt** [comment]
with new 0.109b2 i see this.

Detected I/O inside the event loop. This is causing stability issues. Please report issue to the custom component author for fritzbox_tools doing I/O at custom_components/fritzbox_tools/binary_sensor.py, line 66: self._is_on = status.is_connected
19:58:15 – util/async_.py (WARNING) - il messaggio si è verificato per la prima volta alle 19:56:03 e compare 155 volte

**Comment by mammuth** [comment]
Note: also affects `SwitchDevice`

**Comment by jankatins** [comment]
for the switch I use 

```python
try:
    from homeassistant.components.switch import SwitchEntity
except ImportError:
    # old version which is in my docker currently...
    from homeassistant.components.switch import SwitchDevice as SwitchEntity
```

That way, you don't need to wait for all old users to update to the latest version.

**Comment by mammuth** [comment]
Thanks for the input @jankatins!
This issue was actually solved by #97.

**Comment by N1c093** [comment]
@mammuth wouldn't it be good to close this issue? The problem was solved some month ago.
